cache_ui/advertise.go

```diff
@@ -52,21 +51,37 @@ func (server *CacheServer) CreateAdvertisement(name string, originUrl string, or
 }
 
 func (server *CacheServer) SetFilters() {
+	/*
+	* Converts the list of permitted namespaces to a set and stores it for the serve
+	* This is based on the assumption that the cache server could potentially be filtering once
+	* every minute, so to save speed, we use a map to an empty struct to allow for O(1) lookup time
+	 */
 	server.namespaceFilter = make(map[string]struct{})
-	if viper.IsSet("Cache.AcceptedNamespaces") {
-		nsList := param.Cache_AcceptedNamespaces.GetStringSlice()
-		for _, ns := range nsList {
-			server.namespaceFilter[ns] = struct{}{}
+	nsList := param.Cache_PermittedNamespaces.GetStringSlice()
+	// Ensure that each permitted namespace starts with a "/"
+	for _, ns := range nsList {
+		if ns[0] != '/' {
```

Comment:
Maybe use `strings.HasPrefix`? If someone provides a bad configuration, ns might be empty. I think `strings.HasPrefix` handles being passed an empty string.